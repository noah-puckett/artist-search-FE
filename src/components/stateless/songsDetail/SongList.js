import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Song from '../songsDetail/Song';

function SongList({ songs, artist }) {
    const songsArray = songs.map(song => {
        return (
            <li style={{ listStyleType: 'none' }} key={song.id}>
                <Link to={`/${artist}/${song.title}`}>
                    <Song title={song.title} id={song.id} />
                </Link>
            </li>
        );
    });
    return songsArray;
}

SongList.propTypes = {
    songs: PropTypes.array.isRequired,
};

export default SongList;
