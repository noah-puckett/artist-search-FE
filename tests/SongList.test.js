import React from 'react';
import { shallow } from 'enzyme';
import SongList from '../src/components/stateless/songsDetail/SongList';

describe('SongList component', () => {
    it('renders SongList', () => {
        const wrapper = shallow(<SongList songs={[]} />);
        expect(wrapper).toMatchSnapshot();
    });
});
