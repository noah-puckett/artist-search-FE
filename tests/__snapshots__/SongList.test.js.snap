// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongList component renders SongList 1`] = `null`;
